astropy/units/tests/test_units.py

```diff
@@ -1106,6 +1106,35 @@ def test_si_prefix_symbols(symbol, factor):
     assert np.isclose(base.value / quantity_from_symbol.value, factor, atol=0)
 
 
+@pytest.mark.parametrize(
+    "name,symbol,factor",
+    [
+        pytest.param(name, symbol, factor, id=name)
+        for name, symbol, factor in [
+            ("kibi", "Ki", 2**10),
+            ("mebi", "Mi", 2**20),
+            ("gibi", "Gi", 2**30),
+            ("tebi", "Ti", 2**40),
+            ("pebi", "Pi", 2**50),
+            ("exbi", "Ei", 2**60),
+            # We now switch to float factors because with numpy < 2.0
```

Comment:
This is in a way nicer than the patch from @neutrinoceros, but the disadvantage is that we likely won't notice if our minimum support version for numpy is >= 2.0. So, perhaps write "with NUMPY_LT_2_0"?

On the other hand, probably not really worth running CI again for, so also fine to just merge.

Reply:
Nothing bad will happen if this code is not updated.

Reply:
agreed it's better than the patch I proposed !